### PR TITLE
Fix issues reported by clang-tidy

### PR DESCRIPTION
**Pull request by Danielius1922**

### CODE REVIEW

File: api/oc_helpers.c [hunk 1]
```diff
@@ -288,27 +289,24 @@ oc_conv_hex_string_to_byte_array(const char *hex_str, size_t hex_str_len,
     return -1;
   }
 
-  size_t a = hex_str_len / 2.0 + 0.5;
+  long len = lround((double)hex_str_len / 2.0);
```
Comment (jkralik) [inline api/oc_helpers.c:292]: is it possible to avoid double -> because operations over double is not cheap.

File: security/oc_pstat.c [hunk 2]
```diff
@@ -117,9 +117,6 @@ valid_transition(size_t device, oc_dostype_t state)
       return false;
     break;
   case OC_DOS_RFPRO:
-    if (state == OC_DOS_RFOTM)
-      return false;
-    break;
```
Comment (jkralik) [inline security/oc_pstat.c:122]: why this is removed ? this code seems ok

Comment (Danielius1922) [reply security/oc_pstat.c:122]: It's exactly the same as the next option (`OC_DOS_RFNOP`), so clang-tidy complained about a duplicated case in the switch.

File: api/oc_helpers.c (hunk 1 above)
Comment (Danielius1922) [reply api/oc_helpers.c]: yes we can get rid of it

**Comment by Danielius1922** [comment]
~format

**Review by jkralik — APPROVED** [review]

**Comment by Danielius1922** [comment]
~format

**Pull request merged**